js/windowEventRecorder.js

```diff
@@ -94,7 +94,11 @@
         const textContent = e.target.textContent.trim();
         let textSelector = '';
         if (!e.target.children.length && textContent) {
-          textSelector = `${frameSelector} text=${textContent}`.trim();
+          const isUnique = Array.from(document.querySelectorAll(e.target.tagName)).filter(el => el.textContent.trim() === textContent).length === 1;
```

Comment:
Unfortunately I've seen cases where the duplicates don't necessarily have the same tag name. A common one is `<h1>Login</h1>` and then below there is a `<button>Login</button>`. They click the button in the recording, but then the bots click the earlier one.

Hopefully it's not terribly slow to look at all elements?